### DISCUSSION TITLE
Implementing and using DLL files

### DISCUSSION THREAD

**Discussion by zeller-lorenz** [tauri-apps/tauri#11382] [Q&A] [answered]
Hey!

When I try to run my tauri application I get the error that an specific DLL was not found. (In my case the libcrypto-3-x64.dll)

![image003](https://github.com/user-attachments/assets/05d30d50-8186-4a83-a0fd-774efd9c771f)

My first approach was to implement the DLL Files in my bundle and then load them with the libloading library

```json
"bundle": {
    "resources": [
      "./assets/dll/*"
    ]
}
```

```rust
use libloading::Library;
use tauri::{App, Manager};
use std::path::Path;


pub fn init_dll_files(app: &mut App){

    let resources_dir = app.path().resource_dir().unwrap().as_os_str().to_str().unwrap().to_string();
    let dll_libcrypto_path_string = format!("{}\\{}", resources_dir, "assets\\dll\\libcrypto-3-x64.dll");
    let dll_libcrypto_path = Path::new(&dll_libcrypto_path_string);

    println!("Resource Path: {}", dll_libcrypto_path_string);
    println!("Resource EXISTS: {}", dll_libcrypto_path.exists());

    if dll_libcrypto_path.exists() {
        let lib_libcrypto = Library::new(dll_libcrypto_path).expect("Couldn't load libcrypto-3-x64.dll");
    }

}
```

Is this a possible solution or is there a better way to go?

Thanks in advance!

**Comment by zeller-lorenz** [comment, marked as answer]
I found a workaround:

I copied my libcrypto-3-x64.dll in my src-tauri folder.

In my tauri.config.json I added the resource:

```json
  "bundle": {
    "resources": [
      "./libcrypto-3-x64.dll"
    ]
}
```

After the installation the DLL is automatically in the root directory and the application start works.

**Reply by SilverGhostBS** [reply]
That works perfectly, however this will bundle the .dll for all platforms, is there a way to specify that it should already be bundled in Windows packages (nsis & MSI)?

**Reply by FabianLars** [reply]
you can have a windows specific merge config https://v2.tauri.app/reference/config/#platform-specific-configuration

**Reply by SilverGhostBS** [reply]
That's exactly what I needed! Thanks a lot!
